torch/onnx/_internal/exporter/_compat.py

```diff
@@ -52,52 +57,51 @@ def _from_dynamic_axes_to_dynamic_shapes(
     if input_names is None:
         input_names = []
 
-    sig = _signature(model)
-    if len(input_names) > len(sig.parameters):
-        raise ValueError(
-            f"Number of input names ({len(input_names)}) should not be greater than "
-            f"the number of model inputs ({len(sig.parameters)})"
-        )
-    input_names_to_model_inputs = {}
-    for idx, param_name in enumerate(sig.parameters):
-        if idx < len(input_names):
-            input_names_to_model_inputs[input_names[idx]] = param_name
-        else:
-            input_names_to_model_inputs[param_name] = param_name
+    if kwargs is None:
+        kwargs = {}
 
-    # NOTE: torch.export.export does not support input names assignment,
-    # so we need to map input names to model inputs to create dynamic_shapes
-    # for the exported program
-    dynamic_shapes_to_exported_program = {}
+    dynamic_shapes: dict[str, Any | None] = {}
     for input_name, axes in dynamic_axes.items():
+        # NOTE: torch.export.Dim requires strict min and max constraints, and it
+        # dpends on the traced model to provide the correct min and max values.
+        # We set max to 99999 to avoid the constraints violation error with the default int64 max.
+        # https://github.com/pytorch/pytorch/blob/32f585d9346e316e554c8d9bf7548af9f62141fc/test/export/test_export.py#L687
         if input_name in output_names:
             # User specified an output name as a dynamic axis, so we skip it
             continue
-        # input_name can be either from input_names or from the model inputs
-        if input_name not in input_names_to_model_inputs:
-            raise ValueError(
-                f"dynamic axis: {input_name} is not found in the input names: {input_names}"
-            )
-        model_input_name = input_names_to_model_inputs[input_name]
         if isinstance(axes, dict):
-            dynamic_shapes_to_exported_program[model_input_name] = {
-                k: torch.export.Dim(v) for k, v in axes.items()
+            # Dim needs to pass str.isidentifier()
+            dynamic_shapes[input_name] = {
+                k: torch.export.Dim(re.sub(r"[^A-Za-z_]", "", v), max=99999)
```

Comment:
If the max is not set, llm is going to fail, as sequence length is usually bounded within config. But we also don't want to only support llm. This kind of leaves us with this awkward position.

Reply:
Maybe add this to the source as a reminder?

Reply:
What do you mean the source?

Reply:
source code; add the explanation as a comment in code